code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -238,6 +205,56 @@
 	if(ptank && damage && attack_type == PROJECTILE_ATTACK && prob(15))
 		owner.visible_message("<span class='danger'>[attack_text] hits the fueltank on [owner]'s [src], rupturing it! What a shot!</span>")
 		var/target_turf = get_turf(owner)
-		ignite_turf(target_turf, 100)
+		ignite_turf(target_turf, ptank.air_contents.total_moles(),ptank.air_contents)
 		qdel(ptank)
 		return 1 //It hit the flamethrower, not them
+
+/obj/item/weapon/flamehand
+	name = "burning hand"
+	desc = "Got you to level 3."
+	flags = ABSTRACT
+	w_class = 5
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "fireball"
+
+/obj/item/weapon/flamehand/dropped(user)
+	qdel(src)
+
+/obj/item/weapon/flamehand/afterattack(atom/target, mob/user, flag)
+	if(flag)
+		return
+	var/turf/targetTurf = get_turf(target)
+	var/turfList = getline(user, targetTurf)
+	flame_turf(turfList,null,1,5)
+
+
+
+/atom/proc/flame_turf(turflist, gasmixture, delay, release_amount = 0.75, multiplier = 1)

```

Comment:
This takes a list of turfs? The signature should be better typed, ie `list/turflist`, `datum/gas_mixture/gasmixture`
And it should be plural, `flame_turfs`